_tutorials/2016-05-10-xr-ansible-md.md

```diff
@@ -0,0 +1,204 @@
+git clone git://github.com/ansible/ansible.git --recursive
+cd ansible/ && sudo python setup.py install
+```
+
+Create key to access XR in future:
+```
+vagrant@vagrant-ubuntu-trusty-64:~$ ssh-keygen -t rsa
+Generating public/private rsa key pair.
+Enter file in which to save the key (/home/vagrant/.ssh/id_rsa):
+Enter passphrase (empty for no passphrase):
+Enter same passphrase again:
+.our identification has been saved in
+.pub.public key has been saved in
+The key fingerprint is:
+4e:a5:24:de:e1:51:7c:85:78:a1:65:b1:08:17:e3:be vagrant@vagrant-ubuntu-trusty-64
+The key's randomart image is:
++--[ RSA 2048]----+
+|        ..=o==.  |
+|         =o*+.   |
+|      . + =o.    |
+|     . = *       |
+|      . S .      |
+|       o   .     |
+|        . E      |
+|                 |
+|                 |
++-----------------+
+vagrant@vagrant-ubuntu-trusty-64:~$ 
+vagrant@vagrant-ubuntu-trusty-64:~$ cat ~/.ssh/id_rsa.pub
+ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCfbpi4N3Lcl5i9Y8gd/g4x05IvnIfoPJkhdmGBW2HMFwqWgJQkJF1BM8SuukWeG8+Su4g0Un5tU4/nvbAcqBDR7wFEmB7z7k8VQrXZUxeB4Lc1jEwdDLbxxGOUitLQO+IXlHFJVpkp9Ps6tT82xopaSOQFXKDq0vYXdeEMD/k3NG++0u5pOsJu+kXLIULy1Ix6qvcFDRKbqfT14fU/K7vBYz8gP8Cl+sql9ySm7aOSb0liCBx46/SueBX6uadnMgecVBc1GyvmEX5PwBppUr3Cpby2yOf69iX4NnNcWTTrPY7o7LWuXPNiAgbSAS3R7jVBPyTltB9zCbGq8We5UuVX vagrant@vagrant-ubuntu-trusty-64
+vagrant@vagrant-ubuntu-trusty-64:~$ 
+```
+
+
+
+### XR box pre-configuration
+
+To access XR Linux Shell: 
+> vagrant ssh xr
+
+To access XR console it takes one additional step to figure out port (credentials for ssh: vagrant/vagrant):
+```
+mkorshun@MKORSHUN-2JPYH MINGW64 ~/Documents/workCisco/tutorial
+$ vagrant port xr
+The forwarded ports for the machine are listed below. Please note that
+these values may differ from values configured in the Vagrantfile if the
+provider supports automatic port collision detection and resolution.
+ 22 (guest) = 2223 (host)
+ 57722 (guest) = 2200 (host)
+mkorshun@MKORSHUN-2JPYH MINGW64 ~/Documents/workCisco/tutorial
+$ ssh -p 2223 vagrant@localhost
+vagrant@localhost's password:
+RP/0/RP0/CPU0:ios#
+```
+
+Now let's  configure IP address at XR box. Issue the commant at XR CLI.
+
+```
+conf t
+interface GigabitEthernet0/0/0/0
+ ipv4 address 10.1.1.20 255.255.255.0
+ no shutdown
+!
+commit
+end
+```
+
+Finally, let's check connectivity between boxes:
+```
+RP/0/RP0/CPU0:ios#ping 10.1.1.20
+Mon May  9 08:36:33.071 UTC
+Type escape sequence to abort.
+Sending 5, 100-byte ICMP Echos to 10.1.1.20, timeout is 2 seconds:
+!!!!!
+Success rate is 100 percent (5/5), round-trip min/avg/max = 1/5/20 ms
+RP/0/RP0/CPU0:ios#
+```
+
+
+**Doesn't work for me for now, looking for fix**
+Let's copy public part of key from Ubuntu box and allow access without password. For IOS-XR image 6.1.x commands below:
+```
+run vi /root/.ssh/authorized_keys
+
+run sed -i.bak -e '/^PermitRootLogin/s/no/yes/' /etc/ssh/sshd_config_operns
+
+run service sshd_operns restart
+
+run chkconfig --add sshd_operns
+```
+
```

Comment:
You cannot login to XR as root, violates security.

1)  User vagrant already exists (that is why vagrant ssh works) and sshd_operns has already been added to chkconfig.

2)  User vagrant has sudo privileges already and does NOT require passwords to execute sudo commands.

3) Use the user vagrant for further tasks, copy the ubuntu public key into .ssh/authorized_keys folder under /home/vagrant